### ISSUE TITLE
sql: information_schema tables do not filter users properly

### ISSUE THREAD

**Issue by knz** [cockroachdb/cockroach#35572]
The following tables in info_schema: `table_privileges`, `schemata_privileges`, `role_table_grants` should only show entries pertaining to the current user, not for all users.

Currently they display information for all users.

This can cause breakage in clients that assume that if a permission is visible in these tables, it is necessarily available to the current user.

**Comment by knz** [comment]
Note: even though `schema_privileges` is a crdb extension, it should do the same filtering for consistency with the other two.

**Comment by otan** [comment]
(things like SHOW GRANTS which rely on this info_schema may not work afterwards)

**Comment by knz** [comment]
Huh no what should happen is SHOW GRANTS should be modified accordingly.

Also this statement should only reveal grants of the current user and it's roles, unless the user has an adequate privilege to look up grants for other accounts.


-- 
Verstuurd vanaf mijn Android apparaat met K-9 Mail. Excuseer mijn beknoptheid.

**Comment by otan** [comment]
Yeah I appended this with "this may be intended" after thinking about it.
